src/main/java/com/exasol/sql/expression/rendering/ValueExpressionRenderer.java

```diff
@@ -83,17 +86,28 @@ public void visit(final Function function) {
         if (function.hasParenthesis()) {
             startParenthesis();
         }
-        setLastVisited(function);
     }
 
     @Override
     public void leave(final Function function) {
```

Comment:
Use concrete class as parameter instead of interface. This saves you the `instanceof` trick later.

Reply:
done